frontend/src/components/UserDetails.vue

```diff
@@ -173,13 +185,45 @@ export default {
           if (this.user.attributes.org_details) {
             this.user.attributes.org_details = formatOrganization(this.user.attributes.org_details);
           }
-
           this.$store.commit("user/setUser", this.user);
         })
         .catch(e => {
           console.log(e);
         });
     },
+    loadUserRoles: function(){    
+      this.user.allRoles = [];
+      /*
+      No API call exists to load all roles for a user, so we need to first query
+      for a list of clients, then load the roles for each client one by one
+      */     
+      clients.get().then(allClients => {        
+        allClients.data.forEach(client => {
+            UsersRepository.getUserEffectiveClientRoles(this.userId,client.id).then(clientRoles => {
+              if (clientRoles.data.length>0){
+                let roleCollection = {};
+                roleCollection.clientName = client.name;
+                roleCollection.effectiveRoles = clientRoles.data;
+                this.user.allRoles.push(roleCollection);
+                //Sort results by client name
+                this.user.allRoles.sort(function(a, b) {
+                    var nameA = a.clientName.toUpperCase(); // ignore upper and lowercase
+                    var nameB = b.clientName.toUpperCase(); // ignore upper and lowercase
+                    if (nameA < nameB) {
+                      return -1;
+                    }
+                    if (nameA > nameB) {
+                      return 1;
+                    }
+                    return 0;
+                });
+                //Need to force an update as vue doesn't notice the collection has changed
+                this.$forceUpdate();
```

Comment:
Vue is (understandably?) fussy about how it reacts to state change. Apparently it's more developer friendly in Vue 3. I think you're bumping into the complexity around objects and arrays, see this blog post and the official documentation:

* https://vuejs.org/v2/guide/reactivity.html#Change-Detection-Caveats
* https://michaelnthiessen.com/debugging-guide-why-your-component-isnt-updating/#2-make-sure-to-update-arrays-and-objects-correctly-only-in-vue-2

I recommend making `allRoles`  a root-level `data` property, and then you won't need to call `this.$forceUpdate();`